src/util/fetchUtils.js

```diff
@@ -27,35 +27,15 @@ export const parseQueryResponse = queries => ({ responses, error, message }) =>
 	};
 };
 
-/**
- * Wrapper around `fetch` to send an array of queries to the server. It ensures
- * that the request will have the required OAuth and CSRF credentials and constructs
- * the `fetch` call arguments based on the request method. It also records the
- * CSRF header value in a cookie for use as a CSRF header in future fetches.
- *
- * **IMPORTANT**: This function should _only_ be called from the browser. The
- * server should never need to call itself over HTTP
- *
- * @param {String} apiUrl the general-purpose endpoint for API calls to the
- *   application server
- * @param {Object} options {
- *     method: "get", "post", "delete", or "patch",
- *   }
- * @return {Promise} resolves with a `{queries, responses}` object
- */
-export const fetchQueries = (apiUrl, options) => (queries, meta) => {
-	if (
-		typeof window === 'undefined' &&  // not in browser
-		typeof test === 'undefined'  // not in testing env
-	) {
-		throw new Error('fetchQueries was called on server - cannot continue');
-	}
-	options.method = options.method || 'GET';
+export const getFetchArgs = (apiUrl, options, queries, meta) => {
 	const {
-		method,
 		headers={},
 	} = options;
 
+	const method = (queries[0].meta || {}).method ||  // allow query to set method
+		options.method.toLowerCase() ||  // fallback to options
+		'get';  // fallback to 'get'
```

Comment:
In this case, no, because `import { FOO } from 'bar/baz'` doesn't throw a compile time error when `FOO` doesn't exist but the `'bar/baz'` module _does_ exist, meaning that

```js
import { POSTX } from 'meetup-web-platform/lib/api';

{ method: POSTX }
```

would give you an error at the same time `{ method: 'postx' }` would - it would happen when you first try to send that query to the server at runtime, and the server would tell you that `method` must be one of `'get', 'post', 'delete', 'patch'`.

I _think_ this might be a place where we could set up flow to do compile-time type checking of the arguments to `fetchQueries`, which would probably be better all around without `import` boilerplate.